### ISSUE TITLE
Region problems - changing region doesn't update results

### ISSUE THREAD

**Issue by TravellingGuy** [capitalone/cloud-custodian#12]
I'm giving Cloud Custodian a test run and ran into an issue. I'm not sure if it's a bug or expected behaviour, and if it's expected, how to workaround it.

I'm using the EC2 Old Instance Report (http://www.capitalone.io/cloud-custodian/usecases/usecases/ec2oldinstances.html) for my tests.

On the first run (`custodian run -c ec2-old-instances.yaml -s out`), I got the results I expected. It queried the EC2 resources in the us-east-1 region, filtered them and produced an appropriate output of 12 instances that have been running for more than 60 days.

I then tried to run the same query by specifying a different region (`custodian run --region eu-west-1 -c ec2-old-instances.yaml -s out`) and that didn't work. It simply reused the results from the previous run and the output continued to show the us-east-1 instances.

I deleted the cache file (`~/.cache/cloud-custodian.cache`) and ran it again in the eu-west-1 region and got updated results I expected, EC2 instances in the eu-west-1 region.

I thought if I waited the requisite cache period time (60 seconds by default), it would invalidate the cache, and it would produce updated results when querying on a different region from the previous run, but that didn't happen either.

**Comment by fdosani** [comment]
Hey there,

Can you add the following to the command you are executing `--cache-period=0`
So your call might look like the following:

```
custodian run -c ec2-old-instances.yaml -s out --cache-period=0
```

That's been how I've typically used it.

Thanks,
Faisal

**Comment by kapilt** [comment]
we need to use the region as part of our cache key

**Comment by TravellingGuy** [comment]
Using the -f (--cache) parameter seems to be the ticket. -f (or --cache) allows specifying a different cache file, so specifying a different cache file for each region does what I want.
`custodian run --region us-west-2 --cache us-west-2 -c ec2-old-instances.yaml -s out`
and
`custodian run --region eu-west-1 --cache eu-west-1 -c ec2-old-instances.yaml -s out`
produce expected output for each region.

**Comment by kapilt** [comment]
merged in #18 , added region as part of the cache key for resources. this should obviate the need for separate cache files unless your working with multiple accounts. thanks for the bug report.
